### ISSUE TITLE
Better diagnostic error message when encountering a git-lfs placeholder file

### ISSUE THREAD

**Issue by permcody** [idaholab/moose#17407]
## Reason

If you checkout a repository on a system that doesn't have git-lfs installed or enabled. You will get placeholder files for any files that are stored in git-lfs data stores. Those files contain plain ASCII text like this:
```
version https://git-lfs.github.com/spec/v1
oid sha256:3b7a78871b902e28e46756241a510dcb72de8f3eb2dd3d5bc31342085f1976ed
size 145124
```

The problem is that this confuses some of the readers in MOOSE. In particular we sometimes store large mesh files like this in which case the user will get a message like this:

```
*** ERROR ***
Error opening ExodusII mesh file: /Users/johndoe/projects/foo/problems/big_mesh.e
```

This occurs even if the file is there. 

## Design

We already have utility routines that check for file readability or writeability before trying to interpret data. We are fairly consistent with calling these. I would propose that we add another utility routine that specifically checks for git-lfs placeholder files that we can call from the readability routine. We could then issue a much more useful error and potentially helpful diagnostic message that could help a user recover without seeking assistance from the developer community. 

## Impact

None, moderate annoyance.

**Comment by WilkAndy** [comment]
Yep, this stumped me a while ago.  Not a big deal, but confusing.

**Comment by permcody** [comment]
Closed by #17489
